### PR TITLE
Assert more of FontFallbackMap

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-3109/order-fontfallbackmaplock-#comment-59c9c3d5

### CODE REVIEW

File: third_party/blink/renderer/platform/fonts/font_fallback_list.h [hunk 1]
```diff
@@ -77,6 +77,8 @@ class PLATFORM_EXPORT FontFallbackList : public RefCounted<FontFallbackList> {
   uint16_t Generation() const { return generation_; }
 
   ShapeCache* GetShapeCache(const FontDescription& font_description) {
+    recordreplay::Assert("[RUN-TODO] FontFallbackList::GetShapeCache %d",
```
Comment (Domiii) [inline third_party/blink/renderer/platform/fonts/font_fallback_list.h:80]: This looks like something we definitely want to be aware of.

**Review by bhackett1024 — CHANGES_REQUESTED** [review]
I don't want us sticking in new ordered locks as diagnostics without understanding whether / why they are needed.  Using an ordered lock that's not protecting data isn't useful except in certain specific circumstances as a fence.

**Comment by Domiii** [comment]
> I don't want us sticking in new ordered locks as diagnostics without understanding whether / why they are needed. Using an ordered lock that's not protecting data isn't useful except in certain specific circumstances as a fence.

Hum... If we can't lock - Do we have other tools to check whether a divergence happens due to unordered thread interleavings?
1. I have `Assert`s on all modifications of the data structure.
2. I have `Assert`s on `this` and its owners up-stack.
3. → I think it must be an "unordered access problem", but, short of ordering it, I'm not sure how to test that hypothesis?

File: third_party/blink/renderer/platform/fonts/font_fallback_map.cc [hunk 2]
```diff
@@ -75,6 +76,16 @@ void FontFallbackMap::InvalidateInternal(Predicate predicate) {
       entry.value->MarkInvalid();
     }
   }
+
+  if (recordreplay::IsRecordingOrReplaying() && recordreplay::AreAssertsDisabled()) {
```
Comment (bhackett1024) [inline third_party/blink/renderer/platform/fonts/font_fallback_map.cc:80]: I think the second part of this needs a !

**Review by bhackett1024 — APPROVED** [review]

**Pull request merged**